File: Examples2/objective-c-demo/objective-c-demo/AliPlayerViewController.m
#import "HomeNavViewController.h"
#import "AliPlayerViewController.h"

@interface AliPlayerViewController ()

@end

@implementation AliPlayerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupView];
    
    NSError *error = nil;
    
    [MLYDriver initializeAndReturnError:  &error :^(MLYDriverOptions * option) {
        [[option client] setId: clientID];
        
        MLYMuxModel *muxModel = [self.mlyPlayer getMuxConfigModel];
        [self.plugin adaptWithMuxModel:muxModel];
    }];
    
    if(error != nil){
        NSLog(@"MLYDriver initialize err:%@",error.description);
    }
}
 
- (void)setupView {
    self.plugin = [[MLYAliPlayerPlugin alloc] init];
    CGRect playerViewFrame = CGRectMake(0, 0, UIScreen.mainScreen.bounds.size.width, UIScreen.mainScreen.bounds.size.height);
    UIView *playerView = [[UIView alloc] initWithFrame:playerViewFrame];
    self.mlyPlayer = [[MLYAliPlayer alloc] initWithView:self.view playerView:playerView videoURL:videoURL];
    self.player = self.mlyPlayer.player;
}

- (void)viewDidDisappear:(BOOL)animated {
    [self.plugin deactivate];
    [self.player stop];
    [self.player destroy];
    [super viewDidDisappear:animated];
}

 
@end

File: examples/swift-demo/swift-demo/AliPlayerViewController.swift
import AliyunPlayer
import MLYSDK
import MuxCore
import MuxStatsAliPlayer
import UIKit

class AliPlayerViewController: UIViewController {
    var mlyPlayer: MLYAliPlayer!
    var player: AliPlayer?
    var plugin: MLYAliPlayerPlugin = .init()
     
    override func viewDidLoad() {
        super.viewDidLoad()
        self.setupView()
        do {
            try MLYDriver.initialize { options in
                options.client.id = clientID
            }
            let muxModel = self.mlyPlayer.getMuxConfigModel()
            self.plugin.adapt(muxModel: muxModel)
        } catch {
            print(error)
        }
    }
     
    func setupView() {
        let playerView:UIView =  .init(frame: CGRect(x: 0, y: 0, width: UIView.screenWidth(), height: UIView.screenHeight()))
        self.mlyPlayer = MLYAliPlayer(view: self.view, playerView: playerView, videoURL: videoURL)
        self.player = self.mlyPlayer.player
    }
    
    override func viewDidDisappear(_ animated: Bool) {
        self.plugin.deactivate()
        self.player?.stop()
        self.player?.destroy()
        super.viewDidDisappear(animated)
    }
}
 
